compiler/rustc_middle/src/ty/error.rs

```diff
@@ -908,23 +921,30 @@ fn foo(&self) -> Self::T { String::new() }
         assoc_substs: &[ty::GenericArg<'tcx>],
         ty: Ty<'tcx>,
         msg: &str,
+        is_bound_surely_present: bool,
```

Comment:
Please document this argument's effect. It is clear in the context of this PR, but probably not for anyone just stumbling across it.